### PR TITLE
Added configReload admin task to reload the config, if no deployments are in progress

### PR DESCRIPTION
**Pull request by maheshkelkar**
Also:
- Fixed Deployment hashCode function causing HibernateExpcetion.
- Fixed the enumerated value stored for flow's status in DB

References #44

### CODE REVIEW

File: features/api/config.feature [hunk 1]
```diff
@@ -3,17 +3,25 @@ Feature: DeployDB config APIs
...
-    Then the response should be 204
+    Then the response should be 200
+    And the response body should be:
+    """
+    Done!
+    """
+
 
-  @wip
   Scenario: When system has deployments in progress, attempt to reload config should fail
```
Comment (rtyler) [inline features/api/config.feature]: Can you expand on why this is the case? Also does this mean every deployment must be either COMPLETED or CREATED in order for config reload to work?

File: src/main/groovy/deploydb/ConfigReloadTask.groovy [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    ConfigReloadTask(WorkFlow workFlow) {
+        super('configReload')
+        this.workFlow = workFlow
+    }
+
+    @Timed
+    @Override
+    void execute(ImmutableMultimap<String, String> parameters, PrintWriter output) throws Exception {
+
+        Session session = this.workFlow.deployDBApp.getSessionFactory().openSession()
```
Comment (rtyler) [inline src/main/groovy/deploydb/ConfigReloadTask.groovy]: I don't understand the use of Session or Transactions below, we're not making any database related queries in this process (as far as I can tell/hope), so what's the transaction doing?

File: src/main/groovy/deploydb/ConfigReloadTask.groovy [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        try {
+            ManagedSessionContext.bind(session)
+            Transaction transaction = session.beginTransaction()
+            try {
+                this.workFlow.loadConfigModels(true)
+                transaction.commit()
+                output.println("Done!")
+                output.flush()
```
Comment (rtyler) [inline src/main/groovy/deploydb/ConfigReloadTask.groovy]: Does output need to be flushed? I would *hope* that the caller of execute() would do that for us

**Comment by rtyler** [comment]
> Fixed Deployment hashCode function causing HibernateExpcetion.

In the future, please send notable changes like that in a separate pull request.

File: features/api/config.feature (hunk 1 above)
Comment (maheshkelkar) [inline features/api/config.feature]: BTW, this is no different than commit in #120 

Yes. If every flow (i.e. deployment) is either in SUCCESS or FAILED state. (i.e.  until deployment is verified)

We are mainly protecting config change to affect the verification. We do not want deployment to complete but fail to verify due to config change

File: src/main/groovy/deploydb/ConfigReloadTask.groovy [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+
+/**
+ * Admin task to reload DeployDB configuration
+ */
+class ConfigReloadTask extends Task {
+    private WorkFlow workFlow
+    private static final Logger logger = LoggerFactory.getLogger(ConfigReloadTask.class)
+
+    ConfigReloadTask(WorkFlow workFlow) {
```
Comment (maheshkelkar) [inline src/main/groovy/deploydb/ConfigReloadTask.groovy]: done. will push once everything approved

File: src/main/groovy/deploydb/ConfigReloadTask.groovy (hunk 2 above)
Comment (maheshkelkar) [inline src/main/groovy/deploydb/ConfigReloadTask.groovy]: We cannot UnitOfWork on admin tasks.

We need to check DB for pending deployments/flows

File: src/main/groovy/deploydb/ConfigReloadTask.groovy (hunk 3 above)
Comment (maheshkelkar) [inline src/main/groovy/deploydb/ConfigReloadTask.groovy]: TaskServlet does call output.close(), I am not sure if that calls output.flush(). BTW all the other examples seem to flush(), hence I duplicated it

File: features/api/config.feature (hunk 1 above)
Comment (maheshkelkar) [inline features/api/config.feature]: How about this:

  Scenario: If deployments are NOT verified i.e. are in progress, then attempt to reload
            config should fail. A deployment is considered as in progress, if the status is NOT in SUCCESS or FAILED state

**Comment by rtyler** [comment]
Bouncing back to Mahesh since there are some outstanding items I don't see addressed yet

**Comment by rtyler** [comment]
I'm going to merge this now, but per some of our IRL discussions @maheshkelkar is going to scope out a change in the design for how we manage the relationship between configuration and a Flow

**Pull request merged**